### PR TITLE
Add comprehensive performance benchmarking infrastructure

### PR DESCRIPTION
**Pull request by AbdelStark**
## Summary

This PR introduces a professional-grade performance benchmarking infrastructure for Rustreexo, designed to ensure consistent performance monitoring and regression detection across all critical code paths.

### Comprehensive Benchmark Coverage
- **Stump Operations**: Add/remove operations, verification, memory usage patterns
- **Proof Operations**: Creation, verification, subset operations, updates, memory efficiency
- **Accumulator Comparisons**: Performance analysis across MemForest, Stump, and Pollard implementations

### CI/CD Integration
- **Automated Benchmarks**: Runs on pushes to main branches and performance-related branches
- **Performance Regression Detection**: Compares PR performance against main branch with statistical analysis
- **Continuous Monitoring**: Daily performance tracking with trend analysis and alerting
- **Cross-Platform Testing**: Ubuntu and macOS runners with environment optimization

### Professional Standards
- **Criterion 0.7**: Industry-standard benchmarking framework with HTML reports
- **Deterministic Test Data**: Seeded RNG for reproducible results across runs
- **Statistical Significance**: Multiple iterations with confidence intervals
- **Artifact Management**: 90-day retention for benchmark results and historical data

### Performance Focus
- **Core Operations Only**: Removed serialization benchmarks to focus on algorithmic performance
- **Optimized Parameters**: Balanced execution time with measurement accuracy
- **Environment Tuning**: CPU governor optimization, swap disabled, reduced background processes

### Benchmark Architecture
- Separate benchmark suites for different components
- Parameterized tests across multiple data sizes
- Memory usage and throughput measurements
- Error handling and edge case coverage

### CI Workflow Design
- **Main Benchmark Workflow**: Comprehensive testing with HTML report generation
- **Regression Detection**: PR-triggered performance comparison with baseline
- **Performance Monitoring**: Scheduled daily runs with trend analysis
- **Timeout Management**: 30-minute limits with optimized parameter ranges

### Code Quality Improvements
- Fixed all clippy warnings across the codebase
- Updated to modern Rust patterns (std::hint::black_box)
- Improved format string usage and error handling
- Enhanced code documentation and inline comments

## Benefits

- **Regression Prevention**: Catches performance degradations before they reach production
- **Performance Tracking**: Historical data for optimization decision-making  
- **Development Confidence**: Developers can validate performance impact of changes
- **Professional Standards**: Industry-grade benchmarking practices and reporting

## Testing

All benchmarks compile and run successfully with the new infrastructure. The CI workflows have been tested and optimized for execution time while maintaining statistical accuracy.

## Future Enhancements

This foundation enables future optimizations including:
- Performance budgets for critical operations
- Integration with performance monitoring tools
- Automated optimization suggestions
- Benchmark-driven development workflows

### CODE REVIEW

**Review by Davidson-Souza — COMMENTED** [review]
Strong conceptACK. Just a few comments.

Also, could you squash these for merge?

File: Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ license = "MIT"
 repository = "https://github.com/mit-dci/rustreexo"
 readme = "README.md"
 homepage = "https://github.com/mit-dci/rustreexo"
-rust-version = "1.63.0"
+rust-version = "1.70.0"
```
[the remark this answers is not in the document]
Comment (AbdelStark) [reply Cargo.toml:11]: yes ok

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,247 @@
+name: Performance Benchmarks
+
+on:
+  push:
+    branches: [main, develop, performance/*]
```
[the remark this answers is not in the document]
Comment (AbdelStark) [reply .github/workflows/benchmarks.yml:5]: deleting them
